.circleci/config.yml

```diff
@@ -62,3 +62,77 @@ jobs:
           path: test-reports
           destination: test-reports
 
+      - persist_to_workspace:
+          root: .
+          paths:
+            - ./*
+
+  deploy_to_staging:
+    docker:
+      - image: 18fgsa/cloud-foundry-cli
+        environment:
+          - TZ=America/New_York
+          - CF_API: https://api.fr.cloud.gov
+
+    working_directory: ~/repo
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: deploy Tock Staging to cloud.gov
+          command: |
+            export CF_DEPLOY_USER=$CF_DEPLOYER_USERNAME_STAGING
+            export CF_DEPLOY_PASSWORD=$CF_DEPLOYER_PASSWORD_STAGING
+            cf_deploy.sh tock gsa-18f-tock staging manifest-staging.yml
+
+  deploy_to_production:
+    docker:
+      - image: 18fgsa/cloud-foundry-cli
+        environment:
+          - TZ=America/New_York
+          - CF_API: https://api.fr.cloud.gov
+    working_directory: ~/repo
+
+    steps:
+      - attach_workspace:
+          at: .
+      - run:
+          name: deploy Tock Production to cloud.gov
+          command: |
+            export CF_DEPLOY_USER=$CF_DEPLOYER_USERNAME_PRODUCTION
+            export CF_DEPLOY_PASSWORD=$CF_DEPLOYER_PASSWORD_PRODUCTION
+            cf_deploy.sh tock gsa-18f-tock staging manifest-production.yml
+
+workflows:
+  version: 2
+  build_pull_requests:
+    jobs:
+      - build:
+          filters:
+            branches:
+              only: /.*/
+  build_and_deploy_staging:
+    jobs:
+      - build
+      - deploy_to_staging:
+          requires:
+            - build
+          filters:
+            branches:
+              only: master
+  build_and_deploy_production:
+    jobs:
+      - build:
+          filters:
+            tags:
+              only: /v20[1-9][0-9][0-9]+\.[0-9]+/
+            branches:
+              ignore: /.*/
+      - deploy_to_production:
+          requires:
+            - build
+          filters:
+            tags:
+              only: /v20[1-9][0-9][0-9]+\.[0-9]+/
+            branches:
+              ignore: /.*/
```

Comment:
Should we be explicitly specifying only the `master` branch here too for production?  While this is not a correct use case, it could be possible for someone to accidentally apply a version tag on another branch, and this would pick up on the tag, wouldn't it?

Reply:
We'll have to research this in the future for a couple of reasons:

* What if someone is being intentional about applying a tag elsewhere?
* Circle's docs are a bit ambiguous about only building when a tag is applied and pushed - we're trying to enforce a tag on `master` only here, but it seems like the filters do not interact in the way that we would expect.

As is, this works with the expected behavior, so we'll leave for now!